core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2020 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Obeo - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.sirius.ui.actions;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.sirius.business.api.dialect.command.MoveRepresentationCommand;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.business.api.session.SessionManager;
+import org.eclipse.sirius.business.api.session.danalysis.DAnalysisSelector;
+import org.eclipse.sirius.business.api.session.danalysis.DAnalysisSelectorService;
+import org.eclipse.sirius.business.api.session.danalysis.DAnalysisSession;
+import org.eclipse.sirius.diagram.business.api.query.EObjectQuery;
+import org.eclipse.sirius.ext.base.Option;
+import org.eclipse.sirius.viewpoint.DAnalysis;
+import org.eclipse.sirius.viewpoint.DRepresentationDescriptor;
+import org.eclipse.sirius.viewpoint.ViewpointPackage;
+import org.eclipse.ui.actions.BaseSelectionListenerAction;
+import org.polarsys.capella.core.model.handler.helpers.RepresentationHelper;
+import org.polarsys.capella.core.sirius.ui.Messages;
+
+/**
+ * The action allowing to move representations to a different {@link DAnalysis}.
+ * 
+ * @author <a href="mailto:steve.monnier@obeo.fr">Steve Monnier</a>
+ */
+public class MoveRepresentationsAction extends BaseSelectionListenerAction {
+
+  /**
+   * Constructs the action allowing to open representations.
+   */
+  public MoveRepresentationsAction() {
+    super(Messages.MoveRepresentationsAction_move);
+  }
+
+  @Override
+  protected boolean updateSelection(IStructuredSelection selection) {
+    List<DRepresentationDescriptor> validRepresentationDescriptors = getValidRepresentationDescriptors(selection);
```

Comment:
descriptors shall also be from the same session, otherwise the run doesn't seems to work unlike MoveRepresentationMenuManager